libcoap/src/resource.rs

```diff
@@ -313,6 +314,30 @@ impl<D: Any + ?Sized + Debug> CoapResource<D> {
         }
     }
 
+    /// Adds an attribute with a name and an optional value to a CoAP resource object.
+    /// # Safety
+    /// libcoap uses the passed pointers "attr_name" and "attr_val" here.
+    /// Because of the passed flags, libcoap takes ownership of these pointers.
+    /// Passing NULL for the value is safe, since libcoap only frees the value, if it's not NULL.
```

Comment:
Same as above, safety comments in safe methods should be put directly above the unsafe block.